messaging/WFLY-10518_artemis_no_local_broker.adoc

```diff
@@ -0,0 +1,168 @@
+= [WFLY-10518] Connections to remote Artemis broker must not require local Artemis instance
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          messaging,jms,openshift
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Connections to remote Artemis-based broker (including A-MQ 7) are handled in
+WildFly by the
+https://wildscribe.github.io/WildFly/13.0/subsystem/messaging-activemq/server/pooled-connection-factory/index.html[pooled-connection-factory]
+resource that is in the `messaging-activemq`.
+
+This `pooled-connection-factory` resource can be used to connect to the local
+ Artemis server that "owns" it or to a remote Artemis broker.
+
+However, this `pooled-connection-factory` resource always require the presence
+ of a local Artemis server even when it connects only to a remote Artemis broker.
+This use case will become more prevalent as WildFly is deployed on OpenShift to
+connect to a remote A-MQ 7 broker.
```

Comment:
EAP classical .zip supports integration with A-MQ 7.
Is this RFE going to be the new way of EAP classical .zip integration with A-MQ 7?

If this is going to be the way for EAP classical .zip integration with A-MQ7 aka 3rd party messaging, we need new RFE for this. 

Currently EAP classical .zip integration with A-MQ7 is done via `/subsystem=messaging/server` and if we change the supported way we are basically dropping feature ...

Reply:
I don't follow your point.

This RFE is not dropping any feature. It adds the feature to be able to connect to a remote A-MQ7 *without running a local broker*.
Currently the integration with A-MQ7 is done with `/subsystem=messaging/server/pooled-connection-factory`. With this RFE, it can be done instead with  `/subsystem=messaging/pooled-connection-factory` (no local server required).

Reply:
My point is that we would have 2 ways now for classical .zip

For 3rd party messaging scope we cover `/subsystem=messaging/server` way.
For new way in 3rd party messaging scope we would require new RFE, this is one is OpenShift related and I would like to keep it that way.

Supporting 2 ways on classical .zip has some price, especially on HW load/execution time ++ impact to CP stream.

Reply:
AIUI and IMHO this would be the recommended way on bare metal as well. I don't see anything specific to OpenShift here beyond an offhand reference to using fewer resources there.

Running an embedded broker simply to connect to a remote broker is a waste of resources regardless of the operational environment. And running unnecessary software means greater vulnerability to bugs and security issues.

Someone is going to have to test this for it to be used in the images. If it's tested on OS but we tell bare metal customers it's not supported because <???> that's a strange message.

Reply:
I agree that new was should be more efficient, but existing customers witch are using standalone AMQ won't be probably migrating to this if they are not pushed.

>AIUI and IMHO this would be the recommended way on bare metal as well.

It must be explicitly stated in description of the RFE so it gets documented.


If this is going to be the default way, we can refocus our testing and do limited testing of the old way to limit risks.